File: client/src/app/models/order.ts
import { Address } from "./user"

export interface OrderToCreate {
    basketId: string;
    deliveryMethodId: number;
    shipToAddress: Address;
}

export interface Order {
    id: number
    buyerEmail: string
    orderDate: string
    shipToAdress: Address
    deliveryMethod: string
    shippingPrice: number
    orderItems: OrderItem[]
    subtotal: number
    total: number
    status: string
  }
  
  export interface ShipToAdress {
    firstName: string
    lastName: string
    street: string
    city: string
    state: string
    zipcode: string
  }
  
  export interface OrderItem {
    productId: number
    productName: string
    pictureUrl: string
    price: number
    quantity: number
  }

File: Core/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Core.Entities.OrderAggregate;

public class Order : BaseEntity
{
    public Order()
    {
    }

    public Order(IReadOnlyList<OrderItem> orderItems, string buyerEmail, Address shipToAdress, DeliveryMethods deliveryMethod, decimal subtotal, string paymentIntentId) 
    {
        BuyerEmail = buyerEmail;
        ShipToAdress = shipToAdress;
        DeliveryMethod = deliveryMethod;
        OrderItems = orderItems;
        Subtotal = subtotal;
        PaymentIntentId = paymentIntentId;
    }
    public string BuyerEmail { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public Address ShipToAdress { get; set; }
    public DeliveryMethods DeliveryMethod { get; set; }
    public IReadOnlyList<OrderItem> OrderItems { get; set; }
    public decimal Subtotal { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public string? PaymentIntentId { get; set; } 

    public decimal GetTotal() //Automapper knows when a function has the name with Get at the beginning so it maps this to a property called total
    {
        return Subtotal + DeliveryMethod.Price;
    }

}